api/src/main/java/net/kyori/adventure/pointer/Pointers.java

```diff
@@ -60,6 +64,28 @@
     return new PointersImpl.BuilderImpl();
   }
 
+  /**
+   * Gets all {@code pointer} in this collection which have the given value type.
+   *
+   * @param valueType the type class of the pointer
+   * @param <V> the type of the pointer
+   * @return an unmodifiable set of pointers filtered by the given type
+   */
+  @SuppressWarnings("unchecked")
+  default <V> @NotNull Set<Pointer<V>> pointers(final @NotNull Class<V> valueType) {
+    return Collections.unmodifiableSet(pointers().stream()
+      .filter(pointer -> valueType.isAssignableFrom(pointer.type()))
+      .map(pointer -> (Pointer<V>) pointer)
+      .collect(Collectors.toSet()));
+  }
+
+  /**
+   * Gets all {@code pointer} in this pointer collection.
+   *
+   * @return an unmodifiable set of pointers in this pointer collection
+   */
+  @NotNull Set<Pointer<?>> pointers();
```

Comment:
Not sure if this addition should create a breaking change or if this should be made default and return an empty set.
This however could make the behaviour of the method unpredictable if not overwritten.

Reply:
Hmm yes, annoyingly we need to return something here. Trying to work out if we could make this interface non-extendable. Not quite sure...